toolkit/tools/pkg/licensecheck/licensecheck.go

```diff
@@ -0,0 +1,411 @@
+	all = []LicenseCheckResult{}
+	warnings = []LicenseCheckResult{}
+	errors = []LicenseCheckResult{}
+	for _, result := range l.results {
+		all = append(all, result)
+		if result.HasBadResult() || (pedantic && result.HasWarningResult()) {
+			errors = append(errors, result)
+		}
+		if result.HasWarningResult() && !pedantic {
+			warnings = append(warnings, result)
+		}
+	}
+	return all, warnings, errors
+}
+
+// FormatResults formats the results of the search to a string. Results will be ordered as follows:
+// - Packages with warnings only, sorted alphabetically
+// - Packages with errors (and possibly warnings), sorted alphabetically
+// If pedantic is true, warnings will be treated as errors.
+func (l *LicenseChecker) FormatResults(pedantic bool) string {
+	var sb strings.Builder
+	results := sortAndFilterResults(l.results)
+
+	// Print warnings first, but only if they don't also have an error
+	if !pedantic {
+		for _, result := range results {
+			if result.HasWarningResult() && !result.HasBadResult() {
+				sb.WriteString(printWarning(result))
+			}
+		}
+	}
+
+	// Now print the errors
+	for _, result := range results {
+		if result.HasBadResult() || (pedantic && result.HasWarningResult()) {
+			sb.WriteString(printError(result, pedantic))
+			if !pedantic && result.HasWarningResult() {
+				// If pedantic was set, the warning was already printed as an error.
+				// Otherwise print the warning now.
+				sb.WriteString(printWarning(result))
+			}
+		}
+	}
+
+	return sb.String()
+}
+
+func printWarning(result LicenseCheckResult) string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("WARN: '%s' has license warnings:\n", filepath.Base(result.RpmPath)))
+	sb.WriteString(fmt.Sprintf("\tduplicated license files:\n\t\t%s\n", strings.Join(result.DuplicatedDocs, "\n\t\t")))
+	return sb.String()
+}
+
+func printError(result LicenseCheckResult, pedantic bool) string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("ERROR: '%s' has license errors:\n", filepath.Base(result.RpmPath)))
+	if len(result.BadDocs) > 0 {
+		sb.WriteString(fmt.Sprintf("\tbad %%doc files:\n\t\t%s\n", strings.Join(result.BadDocs, "\n\t\t")))
+	}
+	if len(result.BadFiles) > 0 {
+		sb.WriteString(fmt.Sprintf("\tbad general file:\n\t\t%s\n", strings.Join(result.BadFiles, "\n\t\t")))
+	}
+	if pedantic && len(result.DuplicatedDocs) > 0 {
+		sb.WriteString(fmt.Sprintf("\tduplicated license files:\n\t\t%s\n", strings.Join(result.DuplicatedDocs, "\n\t\t")))
+	}
+	return sb.String()
+}
+
+// runLicenseCheckInChroot searches for bad licenses amongst the RPMs mounted into the chroot. This function is meant
+// to be called from inside the chroot's context.
+func (l *LicenseChecker) runLicenseCheckInChroot() (results []LicenseCheckResult, err error) {
+	const searchReportIntervalPercent = 10
+
+	// Find all the rpms in the chroot
+	rpmsToSearchPaths, err := l.findRpmPaths()
+	if err != nil {
+		return nil, fmt.Errorf("failed to walk srpm directory:\n%w", err)
```

Comment:
Why are we saying srpm?

Reply:
Copy/paste of code from my source code search tool, thought I got them all